content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,39 +35,244 @@ are covered here as well as the `System.Diagnostics` API.
 If you prefer to use OpenTelemetry APIs instead of `System.Diagnostics` APIs,
 you can refer to the [OpenTelemetry API Shim docs for tracing](../shim).
 
-## Traces
+## Example app preparation {#example-app}
+
+This page uses a modified version of the example app from
+[Getting Started](/docs/languages/net/getting-started/) to help you learn
+about manual instrumentation.
+
+You don't have to use the example app: if you want to instrument your own app or
+library, follow the instructions here to adapt the process to your own code.
+
+### Dependencies {#example-app-dependencies}
+
+- [.NET SDK](https://dotnet.microsoft.com/download/dotnet) 6+
+
+### Create and launch an HTTP Server
+
+To begin, set up an environment in a new directory called `dotnet-otel-example`. Within that directory, execute following command:
+
+```shell
+dotnet new web
+```
+
+To highlight the difference between instrumenting a library and a standalone
+app, split out the dice rolling into a library file, which then will be
+imported as a dependency by the app file.
+
+Create the library file named `Dice.cs` and add the following code to it:
+
+```csharp
+/*Dice.cs*/
+
+public class Dice
+{
+    private int min;
+    private int max;
+
+    public Dice(int min, int max)
+    {
+        this.min = min;
+        this.max = max;
+    }
+
+    public List<int> rollTheDice(int rolls)
+    {
+        List<int> results = new List<int>();
+
+        for (int i = 0; i < rolls; i++)
+        {
+            results.Add(rollOnce());
+        }
+        
+        return results;
+    }
+
+    private int rollOnce()
+    {
+        return Random.Shared.Next(min, max + 1);
+    }
+}
+
+```
+
+Create the app file `DiceController.cs` and add the following code to it:
+
+```csharp
+/*DiceController.cs*/
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
+
+
+public class DiceController : ControllerBase
+{
+    private ILogger<DiceController> logger;
+
+    public DiceController(ILogger<DiceController> logger)
+    {
+        this.logger = logger;
+    }
+
+    [HttpGet("/rolldice")]
+    public List<int> RollDice(string player, int? rolls)
+    {
+        if(!rolls.HasValue)
+        {
+            logger.LogError("Missing rolls parameter");
```

Comment:
Avoid using extension methods from LoggerExtension, use [compile-time logging source generation](https://docs.microsoft.com/dotnet/core/extensions/logger-message-generator) pattern to achieve the best performance.

Check https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/logs#use-logger

Reply:
While in general I agree, I don't think that's a widespread enough methodology at the moment, and would take away from what this example is trying to achieve.

Using `ILogger` in the way this example does will be familiar to enough people that it won't cause them to learn something new.

Reply:
@reyang could we address the use of OpenTelemetry .NET Log practices in the Logs section, which will be added in another PR?